src/cohortManager.js

```diff
@@ -0,0 +1,189 @@
+import Cohort from './cohort.js'
+import Student from './student.js'
+
+class CohortManager {
+  constructor() {
+    this.cohorts = []
+    this.id = 1
+  }
+
+  validateCohortName(cohortName) {
+    if (typeof cohortName !== 'string') {
+      throw new Error("Please input a valid name e.g. 'cohort-11'")
+    }
+  }
+
+  validateStudent(student) {
+    if (
+      !student.firstName ||
+      !student.lastName ||
+      !student.github ||
+      !student.email
+    ) {
+      throw new Error(
+        "Please input a valid student e.g. {firstName: 'John', lastName: 'Doe', github: '@johndoe', email:'johndoe@mail.com'}"
+      )
+    }
+  }
+
+  createCohort(cohortName) {
+    this.validateCohortName(cohortName)
+
+    const duplicate = this.cohorts.some((cohort) => cohort.name === cohortName)
+
+    if (!duplicate) {
+      const cohort = new Cohort(cohortName)
+      this.cohorts.push(cohort)
+
+      return this.cohorts.find((cohort) => cohort.name === cohortName)
+    } else {
+      return `${cohortName} already exists`
```

Comment:
This right here. If I'm using your createCohort function, I need to have 3 edge cases accounted for:

- What if an error is thrown
- What if a cohort object is returned
- What if a string is returned

I'd rather this either throw an error or give me a cohort, then I can simply wrap the call in a try / catch